src/steps/users/converter.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  createIntegrationEntity,
+  Entity,
+} from '@jupiterone/integration-sdk-core';
+
+import { Entities } from '../constants';
+import { HexnodeUser } from '../../types';
+
+export function createUserEntity(user: HexnodeUser): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: user,
+      assign: {
+        _type: Entities.USER._type,
+        _class: Entities.USER._class,
+        _key: `${user.name}-${user.id.toString()}`,
+        id: user.id.toString(),
+        username: user.name,
+        // email: user.email, What is the value when null?
```

Comment:
The value of the `email` property in that case would be null (or undefined if it's not returned seeing how it's optional) and that's okay for entity property. I'd add it.

```suggestion
        email: user.email
```

Reply:
There are cases where the user doesn't have an email i.e. the email field returns null.

`{
  id: 1,
  name: 'Default User',
  email: null,
  phoneno: null,
  domain: 'local',
  total_devices: 1
}`

In these cases, the SDK errors out because it seems like an email field in the User entity can not be undefined.

`
  [
    {
      "instancePath": "/email",
      "schemaPath": "#/allOf/1/properties/email/type",
      "keyword": "type",
      "params": {
        "type": "string"
      },
      "message": "must be string"
    }
  ]
`

Reply:
Oh I see, you can check if the value is not null and then store it and if it is null, just store undefined instead. That should work with the SDK.

Reply:
Got it working, just added an `|| undefined`